arm9/src/scrcmd_party.c

```diff
@@ -0,0 +1,813 @@
+    struct Pokemon* party_mon = GetPartyMonByIndex(party, mon_slot);
+
+    *ret_ptr = (u16)GetMonData(party_mon, MON_DATA_MOVE1 + move_slot, NULL);
+    return FALSE;
+}
+
+THUMB_FUNC BOOL ScrCmd_GetPartyMonHeldItem(struct ScriptContext* ctx)
+{
+    struct UnkSavStruct80* sav_ptr = ctx->unk80;
+    u16* ret_ptr = GetVarPointer(ctx->unk80, ScriptReadHalfword(ctx));
+    u16 mon_slot = VarGet(ctx->unk80, ScriptReadHalfword(ctx));
+    struct PlayerParty* party = SavArray_PlayerParty_get(sav_ptr->saveBlock2);
+    struct Pokemon* party_mon = GetPartyMonByIndex(party, mon_slot);
+
+    *ret_ptr = (u16)GetMonData(party_mon, MON_DATA_HELD_ITEM, NULL);
+    return FALSE;
+}
+
+THUMB_FUNC BOOL ScrCmd_ResetPartyMonHeldItem(struct ScriptContext* ctx)
+{
+    struct UnkSavStruct80* sav_ptr = ctx->unk80;
+    u16 mon_slot = VarGet(ctx->unk80, ScriptReadHalfword(ctx));
+    struct PlayerParty* party = SavArray_PlayerParty_get(sav_ptr->saveBlock2);
+    struct Pokemon* party_mon = GetPartyMonByIndex(party, mon_slot);
+
+    u16 party_mon_held_item = ITEM_NONE;
+    SetMonData(party_mon, MON_DATA_HELD_ITEM, &party_mon_held_item);
+
+    return FALSE;
+}
+
+THUMB_FUNC BOOL ScrCmd_CheckPartyForSpecies(struct ScriptContext* ctx)
+{
+    struct UnkSavStruct80* sav_ptr = ctx->unk80;
+    u16* ret_ptr = GetVarPointer(ctx->unk80, ScriptReadHalfword(ctx));
+    u16 species = VarGet(ctx->unk80, ScriptReadHalfword(ctx));
+    u8 party_count = (u8)GetPartyCount(SavArray_PlayerParty_get(sav_ptr->saveBlock2));
+
+    u8 i;
+    for (i = 0, *ret_ptr = 0; i < party_count; i++)
+    {
+        struct PlayerParty* party = SavArray_PlayerParty_get(sav_ptr->saveBlock2);
+        struct Pokemon* party_mon = GetPartyMonByIndex(party, i);
+
+        BOOL party_mon_is_egg = (BOOL)GetMonData(party_mon, MON_DATA_IS_EGG, NULL);
+        if (party_mon_is_egg == FALSE)
+        {
+            u16 party_mon_species = (u16)GetMonData(party_mon, MON_DATA_SPECIES, NULL);
+            if (species == party_mon_species)
+            {
+                *ret_ptr = 1;
+                break;
+            }
+        }
+    }
+
+    return FALSE;
+}
+
+THUMB_FUNC BOOL ScrCmd_CountPartyMonRibbons(struct ScriptContext* ctx)
+{
+    u16* ret_ptr = GetVarPointer(ctx->unk80, ScriptReadHalfword(ctx));
+    u16 mon_slot = VarGet(ctx->unk80, ScriptReadHalfword(ctx));
+    struct PlayerParty* party = SavArray_PlayerParty_get(ctx->unk80->saveBlock2);
+    struct Pokemon* party_mon = GetPartyMonByIndex(party, mon_slot);
+
+    u16 ribbon_idx;
+    u16 ribbons;
+    for (ribbon_idx = 0, ribbons = 0; ribbon_idx < 80; ribbon_idx++)
+    {
+        BOOL party_mon_has_ribbon = (BOOL)GetMonData(party_mon, FUN_0207FC5C((u8)ribbon_idx, 0), NULL);
+        if (party_mon_has_ribbon)
+        {
+            ribbons++;
+        }
+    }
+
+    *ret_ptr = ribbons;
+    return FALSE;
+}
+
+THUMB_FUNC BOOL ScrCmd_Unk022F(struct ScriptContext* ctx)

```

Comment:
I assume this func is supposed to get the total number of types of ribbons held by the party